src/drt/src/dr/FlexDR_maze.cpp

```diff
@@ -1158,6 +1163,24 @@ void FlexDRWorker::modAdjCutSpacingCost_fixedObj(const frDesign* design,
           if (currDistSquare < reqDistSquare)
             hasViol = true;
         }
+        if (!hasViol) {
+          for (auto con : cutLayer->getLef58CutSpacingConstraints()) {
+            reqDistSquare = con->getCutSpacing();
+            reqDistSquare *= con->getCutSpacing();
+            currDistSquare = con->isCenterToCenter() ? c2cSquare : distSquare;
+            if (con->isSameNet()) {
+              continue;
+            }
+            if (con->isParallelOverlap()) {
+              if (prl > 0 && currDistSquare < reqDistSquare) {
+                hasViol = true;
+              }
+            }
+            if (hasViol) {
+              break;
+            }
+          }
+        }
```

Comment:
This neglects many options of the rule which could lead to false positive costs. For example if the rule specifies a restriction on the upper/lower cut layer not the same layer. I would prefer if we remove the maze cost updates from this PR and have another PR that handles all such cases.